File: src/main/java/unificado/core/service/model/Cuenta.java
package unificado.core.service.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Data
@Entity
@Table(name = "plan")
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Cuenta extends Auditable {

    @Id
    @Column(name = "pla_cuenta")
    private BigDecimal numeroMaestro;

    @Column(name = "pla_cuentamaster")
    private BigDecimal numeroMaster;

    @Column(name = "pla_nombre")
    private String nombre;

    @Column(name = "pla_integra")
    private Byte integradora;

    @Column(name = "pla_grado")
    private Integer grado;

    @Column(name = "pla_grado1")
    private BigDecimal grado1;

    @Column(name = "pla_grado2")
    private BigDecimal grado2;

    @Column(name = "pla_grado3")
    private BigDecimal grado3;

    @Column(name = "pla_grado4")
    private BigDecimal grado4;

    @Column(name = "clave")
    private Long cuentaId;

}

File: src/main/java/unificado/api/rest/kotlin/model/Cuenta.kt
package unificado.api.rest.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "plan")
data class Cuenta(

    @Id
    @Column(name = "pla_cuenta")
    var numeroMaestro: BigDecimal? = null,

    @Column(name = "pla_cuentamaster")
    var numeroMaster: BigDecimal? = null,

    @Column(name = "pla_nombre")
    var nombre: String = "",

    @Column(name = "pla_integra")
    var integradora: Byte = 0,

    @Column(name = "pla_grado")
    var grado: Int = 0,

    @Column(name = "pla_grado1")
    var grado1: BigDecimal = BigDecimal.ZERO,

    @Column(name = "pla_grado2")
    var grado2: BigDecimal = BigDecimal.ZERO,

    @Column(name = "pla_grado3")
    var grado3: BigDecimal = BigDecimal.ZERO,

    @Column(name = "pla_grado4")
    var grado4: BigDecimal = BigDecimal.ZERO,

    @Column(name = "clave")
    var cuentaId: Long? = null

) : Auditable()
